ykfr/src/lib.rs

```diff
@@ -324,14 +327,23 @@ impl FrameReconstructor {
                         };
                     }
                     SMLocation::Indirect(reg, off, size) => {
-                        if i == 0 {
-                            // skip first frame
-                            continue;
-                        }
                         debug_assert_eq!(*reg, RBP_DWARF_NUM);
-                        let temp = unsafe { rbp.offset(isize::try_from(*off).unwrap()) };
+                        let temp = if i == 0 {
+                            // While the bottom frame is already on the stack and doesn't need to
+                            // be recreated, we still need to copy over new values from the JIT.
+                            // Luckily, we know the address of the bottom frame, so we can write
+                            // any changes directly to it from here.
+                            unsafe { btmframeaddr.offset(isize::try_from(*off).unwrap()) }
+                        } else {
+                            unsafe { rbp.offset(isize::try_from(*off).unwrap()) }
+                        };
                         debug_assert!(*off < i32::try_from(rec.size).unwrap());
+                        // FIXME: The minimum size reported by the stackmap is 1 which represents 1
+                        // byte. LLVM IR allows for smaller sizes, e.g. `i1` representing a single
+                        // bit. It is currently unclear how that affects this code, so I'm leaving
+                        // this comment here so we don't forget.
```

Comment:
I think LLVM would use a whole byte to store an `i1`.

In theory you could pack a load of `i1`s into a byte, but then it raises questions about what it means to take the address of the bits...

Reply:
I think we should leave the comment anyway.